src/array.rs

```diff
@@ -429,39 +441,42 @@ impl<T: TypeNum, D: Dimension> PyArray<T, D> {
         }
     }
 
-    /// Get the immutable view of the internal data of `PyArray`, as slice.
+    /// Returns the immutable view of the internal data of `PyArray` as slice.
+    /// Please consider the use of [`PyReadOnlyArray::as_slice`](../struct.PyReadOnlyArray.html)
+    /// instead of this.
     ///
-    /// Returns `ErrorKind::NotContiguous` if the internal array is not contiguous.
-    /// # Example
-    /// ```
-    /// use numpy::{PyArray, PyArray1};
-    /// use pyo3::types::IntoPyDict;
-    /// let gil = pyo3::Python::acquire_gil();
-    /// let py = gil.python();
-    /// let py_array = PyArray::arange(py, 0, 4, 1).reshape([2, 2]).unwrap();
-    /// assert_eq!(py_array.as_slice().unwrap(), &[0, 1, 2, 3]);
-    /// let locals = [("np", numpy::get_array_module(py).unwrap())].into_py_dict(py);
-    /// let not_contiguous: &PyArray1<i32> = py
-    ///     .eval("np.arange(10)[::2]", Some(locals), None)
-    ///     .unwrap()
-    ///     .downcast()
-    ///     .unwrap();
-    /// assert!(not_contiguous.as_slice().is_err());
-    /// ```
-    pub fn as_slice(&self) -> Result<&[T], NotContiguousError> {
+    /// # Safety
+    /// If the internal array is not readonly and can be mutated from Python code,
+    /// holding the slice might cause an undefined undefined behavior.
```

Comment:
It really really undefined, huh? 😄 

```suggestion
    /// holding the slice might cause undefined behavior.
```

Reply:
Thanks, it was badly copy-pasted